### PR TITLE
wait for serialized clientId when loading with stashed ops

### LINKED ISSUE #9240: Container should wait for leave message from previous clientId when loaded with stashed ops
## Work Item
**Describe the outcome you expect**

When a container is loaded with stashed ops, it should treat the clientId in the pendingLocalState (if present) the same as the previous clientId upon normal disconnect/reconnect flow, i.e., if the clientId is still present in the quorum when the container sees its own join op, it should wait for a leave message before transitioning to "connected" state.

**Additional context**

#7594
#8247
See #4399 for the general problem of delayed ops on the wire resulting in duplicate op transmission

### PR DESCRIPTION
**Pull request by wes-carlson**
close #9240
Wait for a leave message from the previously serialized container's clientId before transitioning to "connected" state when loading with stashed ops.

In #9242 we will no longer see the leave op, so we will need to cancel the timer at "connect" event if the serialized clientId is not in the quorum.

### CODE REVIEW

File: packages/loader/container-loader/src/connectionStateHandler.ts [hunk 1]
```diff
@@ -65,6 +67,10 @@ export class ConnectionStateHandler {
                 this.applyForConnectedState("timeout");
             },
         );
+        // if we have a serialized clientId we will need to wait unless we see its "leave" op
+        if (serializedClientId !== undefined) {
```
Comment (vladsud) [inline packages/loader/container-loader/src/connectionStateHandler.ts:71]: While connection state manager needs to receive new info (old clientId) in constructor, I believe adding new property makes it more complex and thus more error prone. Constructor should set up object as if we just disconnected (same way as do to right here).
But we should use existing properties only (including this.clientId) and move to (setup) some known existing state that already handles this condition, not create more states and reason about more complex state machine.
In other words, there should not be this.serializedClientId IMHO.

+ @markfields, who is poking at this code right now.

**Comment by vladsud** [comment]
It's not enough to have serialized client ID at that level. We have to restore all layers to exactly same state as if no serialization / rehydration happened.

For example, container runtime needs to set clientId to the one we had such that it can reason about incoming ops, if they are "local" and thus have to be acked. Same for Container layer, who (I believe) calculates local flag (though I'm sure that's not the only thing we depend on client Id).

It would be great to see this process unfold in one PR, even if we want to break it into multiple small PRs. And have UTs that test each individual layer, but also end-to-end tests that validate receiving ops from old client after rehydration.

**Comment by heliocliu** [comment]
Does this change need to go into the 0.58 release?  If so, please merge it before Wednesday and let me know so we can copy it over.  Otherwise please merge this into the `next` branch instead because it is a breaking change.
